### PR TITLE
Plans redesign: Available and missing features

### PR DESCRIPTION
**Pull request by niranjan-uma-shankar**
#### Changes proposed in this Pull Request

This PR implements the experiment that redesigns the signup flow plans step. Users in the treatment group will see the new feature comparison section for each plan in the signup plans step.

NOTE: 
1. This PR uses the changes in https://github.com/Automattic/wp-calypso/pull/50005 as the base branch and is built on top of it.

**Important links**
1. [Figma design](https://www.figma.com/file/gnYBUOBthTMckJRJr3pZmk/Control-Experiments?node-id=1362%3A813)
2. pau2Xa-2yE-p2 for the project thread.

**Annual plans view with free subdomain selected (the first feature item is a generic free domain message)**

calypso localhost_3000_start_plans (5)

**Annual plans view with a custom domain selected (the first feature item mentions the selected domain)**

calypso localhost_3000_start_plans (7)

**Monthly plans view with a free subdomain selected (Live chat and free domain mentioned as unavailable)**

calypso localhost_3000_start_plans (6)

**Monthly plans view with a custom domain selected (Live chat and free domain mentioned as unavailable)**

calypso localhost_3000_start_plans (8)

#### Testing instructions

* Open http://calypso.localhost:3000/ and assign yourself to treatment of the `signup_plans_step_redesign_v1` experiment in the Abacus tool.
* Begin signup at `/start` and verify that the plans step UI matches the [Figma design](https://www.figma.com/file/gnYBUOBthTMckJRJr3pZmk/Control-Experiments?node-id=1362%3A813). Verify for both free subdomain and custom domain selections.
* Verify that the UI matches the Figma design for both annual and monthly plan views.
* Confirm that in Calypso's /plans view, the UI is unchanged.
* Also confirm that the launch flow UI is unchanged.
* Verify that control experience is unchanged.

Related to https://github.com/Automattic/wp-calypso/pull/49877

### CODE REVIEW

File: client/lib/plans/features-list.js [hunk 1]
```diff
@@ -253,6 +259,12 @@ export const FEATURES_LIST = {
 			),
 	},
 
+	[ constants.FEATURE_ADVANCED_SEO_EXPANDED_ABBR ]: {
+		getSlug: () => constants.FEATURE_ADVANCED_SEO_EXPANDED_ABBR,
+		getTitle: () => i18n.translate( 'Advanced SEO (Search Engine Optimisation) tools' ),
```
Comment (a8ci18n) [inline client/lib/plans/features-list.js:264]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **19** times:
`translate( 'Advanced SEO (Search Engine Optimization) tools' )` *ES Score: 6*

File: client/lib/plans/features-list.js [hunk 2]
```diff
@@ -542,6 +560,14 @@ export const FEATURES_LIST = {
 			),
 	},
 
+	[ constants.FEATURE_INSTALL_PLUGINS ]: {
+		getSlug: () => constants.FEATURE_INSTALL_PLUGINS,
+		getTitle: () =>
+			i18n.translate(
+				'Access to more than 50,000 WordPress plugins to extend functionality for your site'
```
Comment (a8ci18n) [inline client/lib/plans/features-list.js:574]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **18** times:
`translate( 'Install plugins and extend functionality for your site with access to more than 50,000 WordPress plugins' )` *ES Score: 8*

**Comment by ollierozdarz** [comment]
Great job with these, @niranjan-uma-shankar! 🙂 

I've got a few quick comments below, mainly as a FYI. 

**Billing toggle**

@olaolusoga recently created a really nice [billing toggle](https://www.figma.com/proto/NhFdwnznk6LEuVtvc18qRm/LOHP-pricing-test?node-id=4%3A6&viewport=471%2C378%2C0.4404674172401428&scaling=min-zoom) for the Control pricing pages (via the Payments team). It might be worth incorporating that as I'm sure it will increase the usability.

Screen Shot 2021-02-16 at 1 58 17 pm

**Unavailable features**

Because we're showing available and unavailable features in the comparison table below each plan, it could be worth using a more subdued Grey (or opacity) for the unavailable features so it's easier to quickly see the value.

Screen Shot 2021-02-16 at 2 04 42 pm

**'Included with..' labels**

For the 'Included with..' labels, I believe Luna have included these above the strikeout text in New onboarding. It might be worth doing the same so we're keeping consistency. Also, in the same vein, we could use the same font-weight for both labels (the heavier weight is preferable).

Screen Shot 2021-02-16 at 2 06 31 pm

**Domain is free for first year**

Lastly, you may have a better idea of what converts better here, but '_The domain example.blog is free for the first year_' feels like it could be shortened to '_example.blog is free for the first year_'. We also say 'Free for One Year', and it might be good to find some consistency in the wording. 
 
Please let me know how else I can help!

**Comment by niranjan-uma-shankar** [comment]
Thanks for the feedack @ollierozdarz . 

> Billing toggle

Thanks, I'll implement that.

> 'Included with..' labels

I've implemented the suggested changes and updated the screenshots on the PR description.


> Lastly, you may have a better idea of what converts better here, but 'The domain example.blog is free for the first year' feels like it could be shortened to 'example.blog is free for the first year'. 

I'll check with Marketing folks on this but the reason why I went for a longer form "The domain ..." is that we've recently been getting reports of users confusing a free domain to mean that the plan itself is free(related issue in https://github.com/Automattic/dotcom-manage/issues/408).

> Because we're showing available and unavailable features in the comparison table below each plan, it could be worth using a more subdued Grey (or opacity) for the unavailable features so it's easier to quickly see the value.

Great point. I've pushed in a change to fix this and now the page looks like this:

Screenshot 2021-02-16 at 7 36 47 PM


I'd also like to mention here a few points raised by @jaswrks in Slack, and I'd love to hear your take on them. I'm paraphrasing his comments:

* @jaswrks mentions that it's not immediately obvious that features are cascading horizontally and that you get more going from left to right. Could a table grid be helpful here(similar to the large comparison table in wordpress.com/pricing)?
* @jaswrks mentions that the features appear like a wall of text. Either we could increase the contrast between available and missing features(update: this is already done) or we could entirely remove the unavailable features(which is how it's shown in the [Figma design](https://www.figma.com/file/gnYBUOBthTMckJRJr3pZmk/Control-Experiments?node-id=1457%3A5140)).

@ollierozdarz What do you think? My personal take is - (1) I don't have strong opinions here, so I defer to you (2) I think the subdued grey has really helped with the readability and looks okay to me but I'm not opposed to entirely removing the unavailable X marked features if it can improve readability.

**Comment by ollierozdarz** [comment]
> What do you think?  My personal take is - (1) I don't have strong opinions here, so I defer to you (2) I think the subdued grey has really helped with the readability and looks okay to me but I'm not opposed to entirely removing the unavailable X marked features if it can improve readability.

@niranjan-uma-shankar I'm **+1** for removing the unavailable features. I haven't included the unavailable features in my newest iteration of the Reskinned onboarding because I personally felt like there was too much to take in at once. Also, by removing those (now subdued) unavailable features, it may help @jaswrks feedback on the left to right parsing. 

**Example:**
Screen Shot 2021-02-17 at 2 04 15 pm

**Comment by jaswrks** [comment]
Thanks for considering my feedback. Really appreciate. Design example in last screenshot looks amazing. I like the toggle above it as well.

**Comment by dzver** [comment]
When I select a paid domain, the plans page works and looks like a screenshot.

Selecting the free domain, however, leads to a JS error.

signup-plans-js-err

**Comment by niranjan-uma-shankar** [comment]
> Selecting the free domain, however, leads to a JS error.

I've fixed this, could you please check again?

**Review by dzver — APPROVED** [review]
LGTM

I verified the signup flow for a new user, for a new site, the launch-site flow, and also confirmed the js error doesn't appear.

**Pull request merged**